client/post-editor/post-editor.jsx

```diff
@@ -408,6 +409,35 @@ export const PostEditor = React.createClass( {
 		return this.state.isSaveBlocked || ! this.state.isEditorInitialized;
 	},
 
+
+	isGoogleDocType: function( type ) {
+		// types identified for gdocs:
+		// "application/x-vnd.google-docs-image-clip+wrapped"
+		// "application/x-vnd.google-docs-document-slice-clip+wrapped"
+		return /google-docs/.test( type );
+	},
+
+	hasTextBeenCopiedFromGoogleDocs: function( e ) {
+		// As per https://html.spec.whatwg.org/multipage/interaction.html#datatransfer
+		// types should be an array containing the available formats that were set on copy.
+		// Some browsers, as Firefox, return a DOMStringList instead, see
+		// https://developer.mozilla.org/en-US/docs/Web/API/DOMStringList
+		const orgTypes = e.clipboardData.types;
+		let types = [];
+		if ( orgTypes.contains ) {
+			types = Array.from( orgTypes );
```

Comment:
It seems that we can't use `Array.from` yet:
https://kangax.github.io/compat-table/es6/#test-Array_static_methods

Also, cc @aduth because of the only other occurrence in Calypso:
https://github.com/Automattic/wp-calypso/pull/550/files#diff-8978d604510d7d46a9ffd636bfb1b43cR103